holder.yml

```diff
@@ -56,60 +56,90 @@ paths:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
         "501":
-          description: Not implemented
+          description: Service not implemented.
         "500":
-          description: internal error
-  /presentations/submissions:
+          description: Internal server error.
+  /exchanges/{exchange-id}/{transaction-uuid}:
     post:
-      summary: Provide a presentation for a holder to store.
-      operationId: storePresentation
-      description: "Delivery endpoint for a client-holder to provide a presentation in compliance with the Verifiable Presentation Request provided in the Notification call. This server-holder MUST match the 'domain' and 'challenge' to a 'domain' and 'challenge' previously sent in a Notification Response, and not already received. The server-holder MUST verify the Presentation. The server MUST store the Presentation if verification passes."
+      summary: Receives information related to an existing exchange.
+      operationId: receiveExchangeData
+      description:
+        A client can use this endpoint to continue the exchange of information
+        associated with an initiated exchange. A Verifiable Presentation is
+        sent to this endpoint. If the server has received all of the information
+        it needs, a 200 OK is returned with either an empty response or a
+        Verifiable Presentation containing requested credentials by the client.
+        If the server requires more information, it returns a Verifiable
+        Presentation Request. A request that the server cannot understand
+        results in an error.
+      parameters:
+        - $ref: "./components/parameters/path/ExchangeId.yml"
+        - $ref: "./components/parameters/path/TransactionUuid.yml"
       requestBody:
-        description: Details the client provides to the server to help it decide if the presentation should be made.
+        description:
+          A Verifiable Presentation.
         content:
           application/json:
             schema:
-              $ref: "#/components/schemas/StorePresentationRequest"
+              $ref: "#/components/schemas/VerifiablePresentation"
       responses:
-        "202":
-          description: Presentation accepted
+        "200":
+          description: Received data was accepted.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/VerifiablePresentationRequest"
         "400":
-          description: Presentation is malformed
-        "401":
-          description: Presentation did not contain a proof
-        "402":
-          description: Payment required
-        "403":
-          description: Presentation verification failed
-        "425":
-          description: Server is unwilling to risk processing a request that might be replayed
-        "501":
-          description: Not implemented
+          description: Received data is malformed.
```

Comment:
I don't see how this could ever be the case given the api definition says that any data is legal X )

Implementers should be throwing 400 as soon as they have validated the request data according to a schema, per OWASP best practices...

https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html

Reply:
@OR13 wrote:
> I don't see how this could ever be the case

Please see this for a more details explanation with a use case: https://github.com/w3c-ccg/vc-api/pull/258/files#r801201273